contracts/distribution/HAMAMPLPool.sol

```diff
@@ -629,10 +630,21 @@ contract LPTokenWrapper {
 }
 
 contract HAMAMPLPool is LPTokenWrapper, IRewardDistributionRecipient {
+    ///@dev here below is our problem for distro modules - need time function changeover, redefine the inflationary arithmetic 
+    ///@notice 0x0e2298E3B3390e3b945a5456fBf59eCc3f55DA16 is currently the YAM addr; need to changeover before deployment to proper addr
     IERC20 public ham = IERC20(0x0e2298E3B3390e3b945a5456fBf59eCc3f55DA16);
-    uint256 public constant DURATION = 625000; // ~7 1/4 days
-
-    uint256 public starttime = 1597172400; // 2020-08-11 19:00:00 (UTC UTC +00:00)
+    
+    
+    ///@dev the below state variables of DURATION and starttime need to be recalculated if not entirely removed to make room for better
+    /*
+    *inflationary/deflationary function along a timeframe that is yet specified
```

Comment:
Lots of considerations here. Might be better to remove convo / suggestions from code and bring into Discord.